### DISCUSSION TITLE
Object Storage Assignment

### DISCUSSION THREAD

**Discussion by bdsoha** [skyvar/.github#3] [Announcements]
@skyvar/everyone For those who have completed completed the previous assignment, a [new assignment](https://github.com/skyvar/object-strorage) has been published.

**Follow the instructions carefully.**

**Any questions regarding the task should be submitted no later than Monday the 27th @ 15:00.**

**Comment by FridaPshitik** [comment]
do we need to create a new branch for this task? in which name ?

**Reply by bdsoha** [reply]
You can work on this locally until you are ready to submit.
Once complete, create a PR from a `feature/***` branch, like always.

**Comment by FridaPshitik** [comment]
according to point number 7- to use the podman or we can use docker instead?

**Reply by FridaPshitik** [reply]
> according to point number 7- to use the podman or we can use docker instead?

@chanaSlumianski @SaraLitman

**Reply by bdsoha** [reply]
Use `docker`, it is the same.

> **Note** In the future, if you want to indicate a *copy mention*, use `/cc @user` syntax.

**Comment by MiriamRotenshtein** [comment]
We saw that there are two types of docker: docker rootfull and docker rootless. Is it called that we are working on docker rootfull?

**Reply by bdsoha** [reply]
Docker desktop is not *rootless*.

**Comment by FridaPshitik** [comment]
We are unable to find the routing of the local installation of the container 
we created in Docker in order to open it in VSC, we tried different commands like PWD and it didn't work...

**Reply by FridaPshitik** [reply]
we got it ...

**Comment by RivkaBerlinski** [comment]
@ShaindyBorzikowsky @chananaimen 
We tried to run `pip install minio` inside a Python container
But we encountered an error that is apparently related to Netfri's security feature, what can be done?

**Reply by bdsoha** [reply]
No need to install MinIO on your own.
That's part of the reason docker was created.

Use an existing image:
- Official: https://hub.docker.com/r/minio/minio
- Unofficial: https://hub.docker.com/r/bitnami/minio

> **Note**: In the future, if you want to indicate a *copy mention*, use `/cc @user` syntax.

**Comment by EsterWeiss** [comment]
I want to make sure I understood correctly, I create two containers, one for minio and one for Python (where I write the client side) and the containers communicate between them (by docker compose)?

**Reply by bdsoha** [reply]
Correct. The MinIO image should be deployed using an existing image.

If you decide to use `python`, I suggest using the `boto3` package.

**Comment by EsterWeiss** [comment]
Now we tried to 
```shell
run pip install boto3 
```  
inside a Python container
But we encountered an error that is apparently related to Netfri's security feature, what can be done?

**Reply by bdsoha** [reply]
https://github.com/skyvar/registration/issues/6#issuecomment-1471529181

**Comment by EsterWeiss** [comment]
the correct command is:
```bash
pip install --trusted-host pypi.org --trusted-host pypi.python.org --trusted-host files.pythonhosted.org boto3
```
This command bypasses security problems (due to Netfri blocking) Note that replace boto3 with the name of the desired package.

**Comment by bdsoha** [comment]
@skyvar/everyone Any questions or updates?

**Reply by esterlev23** [reply]
Everything is fine.
We would like to receive feedback on the correctness of the code.

**Reply by EsterWeiss** [reply]
We finished the task and uploaded to git

**Reply by bdsoha** [reply]
Who is *we*?

**Reply by esterlev23** [reply]
@MiriamRotenshtein, @RivkaBerlinski , @RivkaZishinski , @esterlev23

**Reply by bdsoha** [reply]
Create a PR for review.

**Reply by EsterWeiss** [reply]
> We finished the task and uploaded to git

@ShaindyBorzikowsky @chananaimen @EsterWeiss

**Reply by SaraLitman** [reply]
> @skyvar/everyone Any questions or updates?
we finished too and created a pull request 
@FridaPshitik @chanaSlumianski @SaraLitman

**Reply by bdsoha** [reply]
@SaraLitman I commented on that PR. Please move the discussion there.

**Comment by bdsoha** [comment]
@EsterWeiss Why did you develop on the `main` branch? You worked on the [assignment](https://github.com/skyvar/self-learning/pull/15) that researched this topic? 😞 

/cc @ShaindyBorzikowsky @chananaimen

**Comment by MalkaLichtenshtein** [comment]
@MalkaLichtenshtein @ChayaHershkovitz 
We recently joined and we are trying the object storage exercise,
We run the command -
docker run -p 9001:9001 -it --rm --entrypoint bash -v %cd%:/app minio/minio
To create a minio container and when we enter to the port in the browser,
 there is an error that it cannot find this page...

**Reply by bdsoha** [reply]
1. Use `/cc` when you want to mentioned an indirect recipient.
2. What is the error?
3. Are you using the correct port?
4. Do you need a volume mapping?

**Reply by MalkaLichtenshtein** [reply]
1. OK
2. the error is "the page isn't working"
3. yes, Localhost:9001
4. We using volume for save the program in file when the docker closes

**Reply by bdsoha** [reply]
@SaraLitman Please have a look.

**Reply by MalkaLichtenshtein** [reply]
Ester Weiss helped us find the command :
docker run -p 9000:9000 -p 9090:9090 --name minio -v ~/minio/data:/data -e "MINIO_ROOT_USER=Chaya" -e "MINIO_ROOT_PASSWORD=Chaya1234" quay.io/minio/minio server /data --console-address ":9090"
and now it's good.

**Comment by MalkaLichtenshtein** [comment]
We want to upload the code to GIT,
But this repository has been archived and is now read-only,
How can we anyway?
/cc @ChayaHershkovitz

**Reply by bdsoha** [reply]
Fixed

**Comment by SaraLitman** [comment]
@skyvar/everyone ,
Please note this important note before implementing the code:
https://github.com/orgs/skyvar/discussions/3#discussioncomment-5438888

**Comment by EsterWeiss** [comment]
@skyvar/everyone Any questions or updates?
